app/auth/register/page.tsx

```diff
@@ -0,0 +1,269 @@
+"use client";
+
+import { Alert } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { Form } from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { yupResolver } from "@hookform/resolvers/yup";
+import axios from "axios";
+import Image from "next/image";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { FaLocationArrow } from "react-icons/fa";
+import { GiEarthAmerica } from "react-icons/gi";
+import { LiaPeopleCarrySolid } from "react-icons/lia";
+import * as yup from "yup";
+
+const schema = yup.object().shape({
+  name: yup.string().required("Username is required!"),
+  email: yup
+    .string()
+    .email("Invalid email format!")
+    .required("Email must be filled in!"),
+  password: yup.string().required("Password must be filled in!"),
+});
+
+export default function RegisterPage() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+  const router = useRouter();
+
+  const handleRegister = async (data: any) => {
+    if (loading) {
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      const response = await axios.post(
+        process.env.NEXT_PUBLIC_API + "/auth/register",
+        data,
+      );
+      const res = await response.data;
+
+      if (res.status) {
+        if (typeof window !== "undefined" && window.localStorage) {
+          localStorage.setItem("token", res.data.token);
+          localStorage.setItem("user", JSON.stringify(res.data.user));
+        }
+        router.push("/home");
+      } else {
+        setErrorMessage("Tidak dapat melakukan registrasi!");
+      }
+    } catch (error) {
+      setErrorMessage("Terjadi kesalahan saat registrasi!");
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
Can you replace this with a placeholder function? We're not yet sure how the shape of our authentication API will look like, so just a simple `console.log` will do for now.

Reply:
Also, you can remove the `axios` import.